Tests/SwiftParserTest/Linkage.swift

```diff
@@ -0,0 +1,130 @@
+import XCTest
+
+final class LinkageTest: XCTestCase {
+  // Assert that SwiftSyntax and SwiftParser do not introduce more link-time
+  // dependencies than are strictly necessary. We want to minimize our link-time
+  // dependencies. If this set changes - in particular, if it grows - consult
+  // a SwiftSyntax maintainer to see if there's a way to avoid adding the
+  // dependency.
+  func testLinkage() throws {
+    for i in 0..<_dyld_image_count() {
+      let name = try XCTUnwrap(_dyld_get_image_name(i))
+      let path = String(cString: name)
+      guard path.hasSuffix("SwiftParserTest") else {
+        continue
+      }
+
+      var baseURL = URL(fileURLWithPath: path)
+      while !baseURL.pathComponents.isEmpty, baseURL.pathExtension != "xctest" {
+        baseURL = baseURL.deletingLastPathComponent()
+      }
+
+      if baseURL.pathComponents.isEmpty {
+        XCTFail("Unable to determine path to enclosing xctest bundle")
+        return
+      }
+
+      baseURL = baseURL.deletingLastPathComponent()
+      let swiftSyntaxURL = baseURL.appendingPathComponent("SwiftSyntax.o")
+      try assertLinkage(of: swiftSyntaxURL) { linkages in
+        XCTAssertEqual(linkages, [
+          .library("-lobjc"),
+          .library("-lswiftCompatibilityConcurrency"),
+          .library("-lswiftCore"),
+          .library("-lswiftDarwin"),
+          .library("-lswiftSwiftOnoneSupport"),
+          .library("-lswift_Concurrency"),
+        ])
+      }
+      let swiftParserURL = baseURL.appendingPathComponent("SwiftParser.o")
+      try assertLinkage(of: swiftParserURL) { linkages in
+        XCTAssertEqual(linkages, [
+          .library("-lobjc"),
+          .library("-lswiftCompatibilityConcurrency"),
+          .library("-lswiftCore"),
+          .library("-lswiftDarwin"),
+          .library("-lswiftSwiftOnoneSupport"),
+          .library("-lswift_Concurrency"),
+        ])
+      }
+    }
+  }
+
+  private enum Linkage: Comparable {
+    case library(String)
+    case framework(String)
+  }
+
+  private func readLoadCommands(in object: URL) throws -> [String] {
+    let result = Process()
+    result.executableURL = try XCTUnwrap(URL(fileURLWithPath: "/usr/bin/xcrun"))
+    result.arguments = [
+      "otool", "-l", object.path,
+    ]
+    let outputPipe = Pipe()
+    let errorPipe = Pipe()
+
+    result.standardOutput = outputPipe
+    result.standardError = errorPipe
+    try result.run()
+    result.waitUntilExit()
+    let outputData = outputPipe.fileHandleForReading.readDataToEndOfFile()
+    let output = String(decoding: outputData, as: UTF8.self)
+    return output.components(separatedBy: .newlines)
+  }
+
+  private func assertLinkage(of object: URL, assertion: ([Linkage]) throws -> Void) throws {
+    var linkages = [Linkage]()
+    var lines = try self.readLoadCommands(in: object).makeIterator()
+    while let line = lines.next() {
+      guard line.starts(with: "Load command") else {
+        continue
+      }
+
+      guard
```

Comment:
Could you add an example load command in as a comment here? I think that would make your string processing code easier to understand.